File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/LocalServiceClient.java
package me.ycdev.android.lib.common.demo.service;

import android.content.Context;
import android.support.annotation.NonNull;

import me.ycdev.android.lib.common.ipc.ServiceClientBase;
import me.ycdev.android.lib.common.utils.ThreadManager;

public class LocalServiceClient extends ServiceClientBase<IDemoService> {
    private static final String SERVICE_NAME = "LocalService";

    public LocalServiceClient(@NonNull Context context) {
        super(context, SERVICE_NAME, ThreadManager.getInstance().localServiceRequestIpcLooper(),
                new LocalServiceConnector(context));
    }

}

File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/LocalServiceClient.kt
package me.ycdev.android.lib.common.demo.service

import android.content.Context
import androidx.annotation.NonNull
import me.ycdev.android.lib.common.ipc.ServiceClientBase
import me.ycdev.android.lib.common.utils.ThreadManager

class LocalServiceClient(@NonNull context: Context) : ServiceClientBase<IDemoService>(
    context,
    SERVICE_NAME,
    ThreadManager.instance.localServiceRequestIpcLooper(),
    LocalServiceConnector(context)
) {
    companion object {
        private const val SERVICE_NAME = "LocalService"
    }
}
